testing/regress/ecl/spray_queue_test.ecl

```diff
@@ -0,0 +1,391 @@
+
+#if (VERBOSE = 1)
+    sprayFixedOut1 := output(sfc1,,NAMED('sprayFixedOut1'));
+#else
+    sprayFixedOut1 := output(sfc1, {result},NAMED('sprayFixedOut1'));
+#end
+
+
+// Spray fixed with valid DFU queue
+sfdst2 := NOFOLD(DATASET([{desprayOutFileName + '_FIX', sprayOutFileName + '_FIX', 'dfuserver_queue', 'Spray fixed with default queue:', ''}], sprayRec));
+sfp2 := PROJECT(NOFOLD(sfdst2), sprayFixed(LEFT));
+sfc2 := CATCH(NOFOLD(sfp2), ONFAIL(TRANSFORM(sprayRec,
+                                 SELF.sourceFileName := desprayOutFileName + '_FIX',
+                                 SELF.destFileName := sprayOutFileName + '_FIX',
+                                 SELF.dfuQueue := 'dfuserver_queue',
+                                 SELF.result := 'Spray fixed with default queue: Fail',
+                                 SELF.msg := FAILMESSAGE
+                                )));
+
+#if (VERBOSE = 1)
+    sprayFixedOut2 := output(sfc2,,NAMED('sprayFixedOut2'));
+#else
+    sprayFixedOut2 := output(sfc2, {result},NAMED('sprayFixedOut2'));
+#end
+
+// Spray fixed with invalid DFU queue
+sfdst3 := NOFOLD(DATASET([{desprayOutFileName + '_FIX', sprayOutFileName + '_FIX', 'bela_queue', 'Spray fixed with wrong queue name:', ''}], sprayRec));
+sfp3 := PROJECT(NOFOLD(sfdst3), sprayFixed(LEFT));
+sfc3 := CATCH(NOFOLD(sfp3), ONFAIL(TRANSFORM(sprayRec,
+                                 SELF.sourceFileName := desprayOutFileName + '_FIX',
+                                 SELF.destFileName := sprayOutFileName + '_FIX',
+                                 SELF.dfuQueue := 'bela_queue',
+                                 SELF.result := 'Spray fixed with wrong queue name: Fail',
+                                 SELF.msg := FAILMESSAGE
+                                )));
+
+#if (VERBOSE = 1)
+    sprayFixedOut3 := output(sfc3,,NAMED('sprayFixedOut3'));
+#else
+    sprayFixedOut3 := output(sfc3, {result},NAMED('sprayFixedOut3'));
+#end
+
+
+
+sequential (
+    //output(dropzonePath, NAMED('dropzonePath')),
+    // Preparation
+    setupCsv,
+    setupXml,
+    setupFix,
+    
+    desprayOutCsv,
+    desprayOutXml,
+    desprayOutFix,
+
+    // Spray tests
+    sprayVariableOut1,
+    sprayVariableOut2,
+    sprayVariableOut3,
+    
+    sprayXmlOut1,
+    sprayXmlOut2,
+    sprayXmlOut3,
+    
+    sprayFixedOut1,
+    sprayFixedOut2,
+    sprayFixedOut3,
+
+    // Clean-up
+    FileServices.DeleteExternalFile('.', desprayOutFileName+'_CSV'),
+    FileServices.DeleteExternalFile('.', desprayOutFileName+'_XML'),
+    FileServices.DeleteExternalFile('.', desprayOutFileName+'_FIX'),
+    
+    FileServices.DeleteLogicalFile(sprayPrepFileName+'_CSV'),
+    FileServices.DeleteLogicalFile(sprayPrepFileName+'_XML'),
+    FileServices.DeleteLogicalFile(sprayPrepFileName+'_FIX'),
+    
+    FileServices.DeleteLogicalFile(sprayOutFileName+'_CSV'),
+    FileServices.DeleteLogicalFile(sprayOutFileName+'_XML'),
+    FileServices.DeleteLogicalFile(sprayOutFileName + '_FIX'),
```

Comment:
Am a bit concerned that each new test is doing a lot and regression to test a very small specific issue.
As you said, it would be worth doing the setup of files uses here (which other tests do too) once in setup and minimizing work done here.
Can you open a separate JIRA to reduce  overhead of spray tests?